flink/v1.20/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkSink.java

```diff
@@ -381,7 +381,7 @@ public Builder toBranch(String branch) {
       return this;
     }
 
-    private <T> DataStreamSink<T> chainIcebergOperators() {
+    private DataStreamSink<Void> chainIcebergOperators() {
```

Comment:
why is this change being introduced?

Reply:
This is related to this discussion: https://github.com/apache/iceberg/pull/11244#discussion_r1787274778. I realize that we don't need to use the legacy, deprecated interface in the `FlinkSink` implementation as the sink part of the chain is not important in this chain of transformations - discarding sink behaves the same when it is used either as legacy SinkFunction or the new, SinkV2 interface.